### PR TITLE
Alnitak (New) - Blackmarket ship with a cannon

### PR DESCRIPTION
**Pull request by MasterReef**
## About the PR
 A new ship with a cannon called the ECS Alnitak! Blackmarket only.

## Why / Balance
I like spaceships and guns and so if i have a spaceship and i have a gun boom! Armed Spaceship!.

## Media

- [ X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

**Changelog**

:cl:
- add: Added the first armed ship in the game, ECS Alnitak!

### CODE REVIEW

**Comment by MasterReef** [comment]
would love for it to be added before artillery update

File: SpaceStation14.sln
Comment (dvir001) [inline SpaceStation14.sln:1]: How to make the game not able to boot anymore in one simple step

Comment (MasterReef) [reply SpaceStation14.sln:1]: wait is that needed?

Comment (dvir001) [reply SpaceStation14.sln:1]: Deleting a critical file in your PR that is need related to your ship is an issue.

Comment (MasterReef) [reply SpaceStation14.sln:1]: how to fix?

Comment (dvir001) [reply SpaceStation14.sln:1]: Download it from repo, commit it.
Also move your shuttle to new folders under _NF

Comment (MasterReef) [reply SpaceStation14.sln:1]: i dont know how to commit it back

**Comment by dvir001** [comment]
You still have changes on it, please download the one from this repo and push it to fix it.

File: SpaceStation14.sln
Comment (MasterReef) [reply SpaceStation14.sln:1]: i did it

Comment (dvir001) [reply SpaceStation14.sln:1]: image

Use the one existing on this repo, you submitted your own one to replace it.

Comment (MasterReef) [reply SpaceStation14.sln:1]: the script looks the same?

Comment (MasterReef) [reply SpaceStation14.sln:1]: problem solved i guess.

**Comment by dvir001** [comment]
Also dont forget to move ship to Resources/Maps/_NF/Shuttles/alnitak.yml

**Comment by MasterReef** [comment]
> > > Also dont forget to move ship to Resources/Maps/_NF/Shuttles/alnitak.yml
> > 
> > 
> > good now?
> 
> image Did you forgot to push?

done and dusted

**Comment by dvir001** [comment]
> > > > Also dont forget to move ship to Resources/Maps/_NF/Shuttles/alnitak.yml
> > > 
> > > 
> > > good now?
> > 
> > 
> > image Did you forgot to push?
> 
> done and dusted

You are killing me

image

Also the ship shipyard yml need to point to new locations.

**Comment by MasterReef** [comment]
> > > > > Also dont forget to move ship to Resources/Maps/_NF/Shuttles/alnitak.yml
> > > > 
> > > > 
> > > > good now?
> > > 
> > > 
> > > image Did you forgot to push?
> > 
> > 
> > done and dusted
> 
> You are killing me
> 
> image
> 
> Also the ship shipyard yml need to point to new locations.

Is it???? finally done??? i want to die!!!

**Comment by MasterReef** [comment]
> > > > > Also dont forget to move ship to Resources/Maps/_NF/Shuttles/alnitak.yml
> > > > 
> > > > 
> > > > good now?
> > > 
> > > 
> > > image Did you forgot to push?
> > 
> > 
> > done and dusted
> 
> You are killing me
> 
> image
> 
> Also the ship shipyard yml need to point to new locations.

aaaalllsooo ive read in one of a ships script and it tells me i need to 2 paths

**Comment by dvir001** [comment]
I fixed the files locations for you.

**Comment by Cheackraze** [comment]
its a funny little ship, but we cant allow gas miners on ships like that. You can instead use the atmosfix markers to make them start pre-filled with a large amount of gas, but the miners would be exploited very badly.

**Comment by MasterReef** [comment]
> @MasterReef could you reply to me the status of this, are you going to fix the issues soon ? Thank you in advance

Sorry for late reply, I will fix it when I can.

**Comment by MasterReef** [comment]
> @MasterReef could you reply to me the status of this, are you going to fix the issues soon ? Thank you in advance

All fixed!

**Review by MagnusCrowe — CHANGES_REQUESTED** [review]
1. The cannon portion of the ship has 2 tiny fans where it technically needs none, due to the airlock...one at the most.  Same issue with the docking ports.
2. Your distro is all passive vents.  In the event of any spacing on your ship, everything will be spaced because of this.
3. The tiny cockpit has no scrubber, that will kill people.
4. Ship starts off with too many canisters of oxygen and plasma.
5. Remove the captain hardsuit locker.
6. Remove the chemical grenade stuff and the cryobeakers.
7. You have a sink under a light in your chem room, move one of the two please.
image
8. Your air alarm and APC should be rotated to face the correct direction.
9. You might want to include an "igniter" and a proper atmospheric setup in your cannon part so nobody will need to sacrifice themselves every time you want to fire.

Lastly, I think that this ship is too dependent on the server atmos settings.  IDK if it will even work with the current speed of atmos. 
image

**Comment by MagnusCrowe** [comment]
Also, you are missing a fax machine.
Please refer to https://frontierstation14.com/index.php/Ship_Submission_Guidelines for anything I may have forgotten.

**Comment by Cheackraze** [comment]
i recently sped atmos up a tad and might do so again since its still not quite where I want it to be @MagnusCrowe. Theoretically, with plasma burn completed in that chamber, it _should_ work as a giant pneumatic cannon

**Comment by MasterReef** [comment]
> 1. The cannon portion of the ship has 2 tiny fans where it technically needs none, due to the airlock...one at the most.  Same issue with the docking ports.
> 2. Your distro is all passive vents.  In the event of any spacing on your ship, everything will be spaced because of this.
> 3. The tiny cockpit has no scrubber, that will kill people.
> 4. Ship starts off with too many canisters of oxygen and plasma.
> 5. Remove the captain hardsuit locker.
> 6. Remove the chemical grenade stuff and the cryobeakers.
> 7. You have a sink under a light in your chem room, move one of the two please.
>    image
> 8. Your air alarm and APC should be rotated to face the correct direction.
> 9. You might want to include an "igniter" and a proper atmospheric setup in your cannon part so nobody will need to sacrifice themselves every time you want to fire.
> 
> Lastly, I think that this ship is too dependent on the server atmos settings. IDK if it will even work with the current speed of atmos. image

In the 2nd pic, you have to put the payload in the barrel, not in the ignition chamber.

I am gonna fix these when I can.

**Comment by MasterReef** [comment]
> 1. The cannon portion of the ship has 2 tiny fans where it technically needs none, due to the airlock...one at the most.  Same issue with the docking ports.
> 2. Your distro is all passive vents.  In the event of any spacing on your ship, everything will be spaced because of this.
> 3. The tiny cockpit has no scrubber, that will kill people.
> 4. Ship starts off with too many canisters of oxygen and plasma.
> 5. Remove the captain hardsuit locker.
> 6. Remove the chemical grenade stuff and the cryobeakers.
> 7. You have a sink under a light in your chem room, move one of the two please.
>    image
> 8. Your air alarm and APC should be rotated to face the correct direction.
> 9. You might want to include an "igniter" and a proper atmospheric setup in your cannon part so nobody will need to sacrifice themselves every time you want to fire.
> 
> Lastly, I think that this ship is too dependent on the server atmos settings. IDK if it will even work with the current speed of atmos. image

Everything you told me is now fixed, if i missed some then tell me.

**Comment by MasterReef** [comment]
> i recently sped atmos up a tad and might do so again since its still not quite where I want it to be @MagnusCrowe. Theoretically, with plasma burn completed in that chamber, it _should_ work as a giant pneumatic cannon

Ooooo thanks, Also do you know a way for cryogenic beakers that is filled with clf3 to explode on impact when thrown at walls from outside the station?

**Comment by MasterReef** [comment]
I will be not available from 1/3 to 7/3. fyi if you guys want me to make changes.

**Comment by MasterReef** [comment]
I am available now

**Comment by MagnusCrowe** [comment]
Today in fact.

**Comment by MasterReef** [comment]
> Today in fact.

Not meaning to be rude but.... is 2 days after the comment still today though?

No offense intended

**Comment by MagnusCrowe** [comment]
You need to update your branch to the latest version on Frontier master, the version you're using is missing a lot of things.

1.  You need to replace the air vent in the burn chamber with a gas injector.
2. You don't need a disposals pipe to transport a lighter to the burn chamber, you can use a device called an igniter.
3. Instead, you can use the disposals pipe to load the cannon without going into space probably.
4. You need to replace the generators with the shuttle variant once your branch is up to date. These ones are automatically anchored, fueled, and running.
5. I recommend you replace the captain's and merc's locker with pirate lockers.
6. You can include a chem dispenser in your chemistry lab if you wish.
7. You don't need an air sensor in rooms with powered vents and scrubbers.  You only need air sensors in rooms without them, you might want to remove the one from the cargo area and add one to the burn chamber, to monitor the pressure and temperature remotely.
8. Link your burn chamber's airlock doors so that opening one bolts the other so they both cannot be opened at the same time.  That would be catastrophic.
9. The gas mixer in your atmos setup has 30% nitrogen and 70% oxygen, the incorrect ratio. It should be about 22% oxygen and 78% nitrogen.
10. You need to add a few emergency lights.
11. Your substation is located on a rear wall right behind your thrusters, a prime location to become EMPed...might reduce the ships value as a contraband vessel.
12. The ship's sell value and purchase price are really far off of eachother, was that intentional?

image

**Comment by MagnusCrowe** [comment]
Forgot a couple things.
1. You may want to add a fire extinguisher and a defib wall closet.
2. I'm confused why you have a passive vent under a wall.
3. Please route your cables under floor tiles and not under walls wherever possible.
image

**Comment by MasterReef** [comment]
> You need to update your branch to the latest version on Frontier master, the version you're using is missing a lot of things.
> 
> 1. You need to replace the air vent in the burn chamber with a gas injector.
> 2. You don't need a disposals pipe to transport a lighter to the burn chamber, you can use a device called an igniter.
> 3. Instead, you can use the disposals pipe to load the cannon without going into space probably.
> 4. You could probably also use disposals to drop grenades out the back of the ship to collide with people who chase you.
> 5. You need to replace the generators with the shuttle variant once your branch is up to date. These ones are automatically anchored, fueled, and running.
> 6. I recommend you replace the captain's and merc's locker with pirate lockers.
> 7. You can include a chem dispenser in your chemistry lab if you wish.
> 8. You don't need an air sensor in rooms with powered vents and scrubbers.  You only need air sensors in rooms without them, you might want to remove the one from the cargo area and add one to the burn chamber, to monitor the pressure and temperature remotely.
> 9. Link your burn chamber's airlock doors so that opening one bolts the other so they both cannot be opened at the same time.  That would be catastrophic.
> 10. The gas mixer in your atmos setup has 30% nitrogen and 70% oxygen, the incorrect ratio. It should be about 22% oxygen and 78% nitrogen.
> 11. You need to add a few emergency lights.
> 12. Your substation is located on a rear wall right behind your thrusters, a prime location to become EMPed...might reduce the ships value as a contraband vessel.
> 13. The ship's sell value and purchase price are really far off of eachother, was that intentional?
> 
> image

Fixed, also the ignition chamber needs to warm up to fire, the pressure inside needs atleast 10-20k kpa.
Ship price is high cuz, you can fire a ton of ordinance in one shot and that can absolutely cut an entire ship in half. (If only chemical explosions work in places without atmos like how rpg works in space. Prob gonna add a new cannon ammo.)

Also theres wires in the in the walls of the barrel because the barrel does not have tiling.

**Comment by MasterReef** [comment]
@MagnusCrowe Can you review this when you can?

**Comment by MagnusCrowe** [comment]
> @MagnusCrowe Can you review this when you can?

Today/Tonight

**Comment by MagnusCrowe** [comment]
You're going to need to update this branch before I do any more reviewing, sorry it's just so out of date.
See sec for example of how old this branch is.
image

**Comment by dvir001** [comment]
Im closing this PR till you are doing a master pull into it, 
you can reopen it or ask us here to reopen it for you.
Or you can also just move the 2 files into a new PR on updated version of your fork.

**Comment by MasterReef** [comment]
I give up. This was here from January 22.

**Pull request closed without merging**